dependencies.yaml

```diff
@@ -126,39 +126,45 @@ dependencies:
       - output_types: conda
         packages:
           - &doxygen doxygen=1.9.1
-  cudatoolkit:
+  cuda_version:
     specific:
       - output_types: conda
         matrices:
           - matrix:
               cuda: "11.2"
             packages:
               - cuda-version=11.2
```

Comment:
Does `rapids-dependency-file-generator` have some concept of templating that we could use?

If so, maybe we could rewrite this to something like and allow that to be reused across different CUDA versions and cutdown on the boilerplate further

```yaml
          - matrix:
              cuda: "{{ cuda }}"
            packages:
              - cuda-version={{ cuda }}
```

Reply:
Currently, there is not support for anything like that. I think we’ve avoided introducing a templating layer on purpose, in part. It’s a little bit like how conda recipes want to move away from template logic… we’ve gotten quite far without it for dependencies, but I see the value.

Reply:
For clarity Conda recipes will still have templating. It's more that we want the recipe format to reflect YAML. The reason being we can read them with a normal YAML library in one pass (not lots of passes through with Jinja, selectors, etc.)

There are ways to have templating and make it YAML friendly. Though this is maybe another discussion for a different issue